src/clj/ataru/applications/excel_export.clj

```diff
@@ -352,22 +352,29 @@
        (util/non-blank-val (:label form-field) [:fi :sv :en])))
 
 (defn- belongs-to-other-hakukohde?
-  [selected-oids form-field]
-  (and (not-empty selected-oids)
-       (let [belongs-to (set (concat (:belongs-to-hakukohderyhma form-field)
-                                     (:belongs-to-hakukohteet form-field)))]
-         (and (not-empty belongs-to)
-              (empty? (clojure.set/intersection selected-oids belongs-to))))))
+  [form-field-belongs-to form-fields-by-id form-field]
+  (defn some-parent-belongs-to-other-hakukohde?
+    [field-with-parent]
+    (when-let [parent (or (some->> field-with-parent
+                                   :children-of
+                                   (get form-fields-by-id))
+                          (some->> field-with-parent
+                                   :followup-of
+                                   (get form-fields-by-id)))]
+      (or (not (form-field-belongs-to parent))
```

Comment:
Tässä voi kutsua `belongs-to-other-hakukohde?` rekursiivisesti

Reply:
Refaktoroin koko funktio.